src/formio/components/EditGrid.stories.jsx

```diff
@@ -268,3 +268,93 @@ export const AddressNLWithNoData = {
     await userEvent.click(saveButton);
   },
 };
+
+export const WithLayoutComponents = {
+  render,
+  args: {
+    key: 'editgrid',
+    label: 'Nested layout components',
+    extraComponentProperties: {
+      hideLabel: false,
+      components: [
+        {
+          type: 'textfield',
+          key: 'text',
+          label: 'A text field',
+        },
+        {
+          type: 'content',
+          key: 'test1',
+          input: false,
+          label: 'WYSIWYG content',
+          html: '<p>Bonjour</p>',
+        },
+        {
+          type: 'fieldset',
+          key: 'level2Group',
+          label: 'Nested fieldset',
+          hideHeader: true,
+          components: [
+            {
+              type: 'textfield',
+              key: 'nestedText',
+              label: 'Nested text',
+            },
+          ],
+        },
+        {
+          type: 'columns',
+          key: 'columns',
+          label: 'Nested columns',
+          columns: [
+            {
+              size: 6,
+              components: [{type: 'number', key: 'number1', label: 'Number 1'}],
+            },
+            {
+              size: 6,
+              components: [{type: 'number', key: 'number2', label: 'Number 2'}],
+            },
+          ],
+        },
+      ],
+      inlineEdit: false,
+      disableAddingRemovingRows: false,
+      addAnother: 'Nog één toevoegen',
+      saveRow: 'Bevestigen',
+      removeRow: 'Verwijderen',
+    },
+
+    data: [
+      {
+        text: 'Some text field value',
+        nestedText: 'Nested text field value',
+        number1: 42,
+        number2: 420,
+      },
+    ],
+  },
+
+  play: async ({canvasElement, step}) => {
+    const canvas = within(canvasElement);
+
+    // we expect the input element labels and values to be displayed
+    expect(await canvas.findByText('A text field')).toBeVisible();
+    expect(await canvas.findByText('Some text field value')).toBeVisible();
+
+    expect(await canvas.findByText('Nested text')).toBeVisible();
+    expect(await canvas.findByText('Nested text field value')).toBeVisible();
+
+    expect(await canvas.findByText('Number 1')).toBeVisible();
+    expect(await canvas.findByText('42', {exact: true})).toBeVisible();
+
+    expect(await canvas.findByText('Number 2')).toBeVisible();
+    expect(await canvas.findByText('420', {exact: true})).toBeVisible();
+
+    await step('Layout component labels not shown', () => {
+      expect(canvas.queryByText('WYSIWYG content')).not.toBeInTheDocument();
```

Comment:
If I understand the problem correctly, a content field in a repeating group showed the label instead of the actual content. I see you added a check that the label 'WYSIWYG content' is no longer visible. Would it be useful to also add a check that the content ('Bonjour!') is actually displayed now?

Reply:
Not really, but ultimately it's a design choice to not display layout components so I'm sure someone at some point will ask for this :laughing: 

For some more context - Formio components can be put roughly in two categories: input components, and layout components. Input components are actual form fields, like a textfield, number, dropdown... Layout components are used to organize the structure of the form, like a fieldset for a set of related inputs, or columns layouts. We support three layout component types: fieldsets, columns and content (WYSIWYG). Layout components never have a value associated with them.

The default behaviour of Formio is to only display input components in this summary - when you edit the item again, layout components are also displayed since they are related to UX. However, the default template from Formio shows data in a tabular format, and that is something we had to override. When we did that override, we didn't take into account that layout components aren't/shouldn't be displayed, and they were treated like input components: display label followed by the value.

Displaying columns/fieldsets here doesn't make any sense at all, for content you could argue that displaying the WYSIWYG could be desired, but I went for the simple approach of just skipping all layout components alltogether.

Reply:
OK, clear